x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/DataFrameDataExtractorFactory.java

```diff
@@ -94,8 +97,30 @@ public static DataFrameDataExtractorFactory createForSourceIndices(Client client
 
     private static TrainTestSplitterFactory createTrainTestSplitterFactory(Client client, DataFrameAnalyticsConfig config,
                                                                            ExtractedFields extractedFields) {
-        return new TrainTestSplitterFactory(client, config,
-            extractedFields.getAllFields().stream().map(ExtractedField::getName).collect(Collectors.toList()));
+        return new TrainTestSplitterFactory(
+            client,
+            config,
+            Stream.concat(Arrays.stream(
+                extractOrganicFeatureNames(extractedFields)),
+                Arrays.stream(extractProcessedFeatureNames(extractedFields))
+            ).collect(Collectors.toList()));
+    }
+
+    static String[] extractOrganicFeatureNames(ExtractedFields extractedFields) {
```

Comment:
I wonder if it would make sense for this and `extractProcessedFeatureNames` to become member methods of `ExtractedFields` and get them unit tested too.

Reply:
Yeah, member methods make sense as they only touch internal data